src/test/scala/com/cognite/sdk/scala/v1/DataSetsTest.scala

```diff
@@ -0,0 +1,96 @@
+package com.cognite.sdk.scala.v1
+import java.time.Instant
+
+import com.cognite.sdk.scala.common._
+import com.cognite.sdk.scala.common.SetValue
+
+class DataSetsTest extends SdkTestSpec {
+
+  it should "able to create datasets if doesn't exists" in {
+    val results =
+      client.dataSets.filter(DataSetFilter(externalIdPrefix = Some("testDataset1-"))).compile.toList
+
+    // We can't just keep on creating because datasets doesn't support deletes yet so we filter and then create.
+    // We can have two datasets with same name but not same externalId.
+
+    results.headOption.getOrElse({
```

Comment:
Doesn't this test only run once, as the externalId is not changing so results will have one item?

Reply:
Yeah, I will just filter it without any parameters and then handle the error if it exists.
We can just keep on inserting because datasets don't suporrt deletes yet.